### ISSUE TITLE
`restrict-plus-operands` incorrect identifies `Date` as `RegExp`

### ISSUE THREAD

**Issue by JakobJingleheimer** [oxc-project/tsgolint#518]
I think there may actually be 2 problems:

* Date is missing from the allow-list.

* The identification is either falling back to `RegExp` or is actually mistaking a `Date` for a `RegExp`.

```ts
''+new Date()
```

> Invalid operand for a '+' operation. Operands must each be a number or string, allowing a string + any of: `any`, `boolean`, `null`, `RegExp`, `undefined`. Got `RegExp'.

**Comment by camc314** [comment]
`restrict-plus-operands` doesn't have an allowlist ref: https://github.com/typescript-eslint/typescript-eslint/blob/63ab00225eea8c9e4c8fadc22d5c3081daf13e18/packages/eslint-plugin/src/rules/restrict-plus-operands.ts#L61-L88

it's correct that we're reporting an error here (we report the same as typescript-eslint

https://typescript-eslint.io/play/#ts=5.9.2&fileType=.tsx&code=OTAEGpQOwUwd1AEQIYBcYAoCUBuIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6aFfKsu4%2BWZLQD2paAEMmAE1QZIiaNCHRI4MAF8QaoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

However the error we're reporting is incorrect:

```
Invalid operand for a '+' operation. Operands must each be a number or string, allowing a string + any of: `any`, `boolean`, `null`, `RegExp`, `undefined`. Got `RegExp`.
```
When we should be reporting:
```
Invalid operand for a '+' operation. Operands must each be a number or string, allowing a string + any of: `any`, `boolean`, `null`, `RegExp`, `undefined`. Got `Date`.
```

<img width="967" height="237" alt="Image" src="https://github.com/user-attachments/assets/6a3e11a4-2d52-4d62-b930-260bb56b1a56" />

**Comment by JakobJingleheimer** [comment]
Why is it correct to report an error? `''+new Date()` is perfectly fine (as are a bunch of others besides string and number).

**Comment by camc314** [comment]
Closing as fixed

younggglcy fixeed reporting a `Date` as a RegExp` in #786 

reporting an error on this code is correct as it aligns with typescript-eslint.

let me know if that's not the case
